internal/dispatch/graph/check_test.go

```diff
@@ -197,6 +197,92 @@ func TestMaxDepth(t *testing.T) {
 	require.Equal(v1.DispatchCheckResponse_UNKNOWN, checkResult.Membership)
 }
 
+func TestCheckMetadata(t *testing.T) {
+	type expected struct {
+		relation              string
+		isMember              bool
+		expectedDispatchCount int
+		expectedDepthRequired int
```

Comment:
Can we come up with a test where expectedDepthRequired and expectedDispatchCount are different values

Reply:
Yeah, I'll add something that dispatches on the same level

Reply:
Something with a TTU with two values would be good.

Reply:
Should add another one? I added one already